src/main/java/ch/njol/skript/expressions/ExprCollidableExemptions.java

```diff
@@ -0,0 +1,131 @@
+import org.bukkit.Bukkit;
+import org.bukkit.entity.Entity;
+import org.bukkit.entity.LivingEntity;
+import org.bukkit.event.Event;
+import org.eclipse.jdt.annotation.Nullable;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+@Name("Collidable Exemptions")
+@Description("Gets/sets/removes entities from the collidable exemptions of a player.")
+@Examples({
+	"send collidable exemptions of player",
+	"add passengers of player and vehicle of player to collidable exemptions of player",
+	"remove target entity from collidable exemptions of player",
+	"reset collidable exemptions of player"
+})
+@Since("INSERT VERSION")
+public class ExprCollidableExemptions extends PropertyExpression<LivingEntity, LivingEntity> {
+
+	static {
+		register(ExprCollidableExemptions.class, LivingEntity.class, "collidable exemptions", "livingentities");
+	}
+
+	@Override
+	protected LivingEntity[] get(Event event, LivingEntity[] source) {
+		List<LivingEntity> entities = new ArrayList<>();
+		for (LivingEntity livingEntity : source){
+			livingEntity.getCollidableExemptions().forEach(uuid -> {
+				Entity entity = Bukkit.getEntity(uuid);
+				if (entity instanceof LivingEntity){
+					entities.add((LivingEntity) entity);
+				}
+			});
+		}
+		return entities.toArray(LivingEntity[]::new);
+	}
+
+	@Override
+	public Class<? extends LivingEntity> getReturnType() {
+		return LivingEntity.class;
+	}
+
+	@Override
+	public String toString(@Nullable Event event, boolean debug) {
+		return "collidable exemptions";
+	}
+
+	@Override
+	public boolean init(Expression<?>[] expressions, int matchedPattern, Kleenean isDelayed, SkriptParser.ParseResult parseResult) {
+		setExpr((Expression<? extends LivingEntity>) expressions[0]);
+		return true;
+	}
+
+	@Nullable
+	@Override
+	public Class<?>[] acceptChange(Changer.ChangeMode mode) {
+		switch (mode) {
+			case ADD:
+			case SET:
+			case REMOVE:
+			case RESET:
+			case DELETE:
+				return CollectionUtils.array(LivingEntity[].class);
+		}
+		return null;
+	}
+
+	@Override
+	public void change(Event event, @Nullable Object[] delta, Changer.ChangeMode mode) {
+		List<UUID> uuidList = new ArrayList<>();
+		if (delta != null){
+			for (Object object : delta){
+				if (object instanceof LivingEntity)
+					uuidList.add(((LivingEntity) object).getUniqueId());
+			}
+		}
+		switch (mode) {
+			case SET:
+			case DELETE:
+			case RESET:
+				getExpr().stream(event).forEach(livingEntity -> {
+					livingEntity.getCollidableExemptions().clear();
+					if (!uuidList.isEmpty()) {
+						livingEntity.getCollidableExemptions().addAll(uuidList);
+					}
+				});
+				break;
+			case ADD:
+				getExpr().stream(event).forEach(livingEntity ->
+					livingEntity.getCollidableExemptions().addAll(uuidList));
+				break;
+			case REMOVE:
+				getExpr().stream(event).forEach(livingEntity ->
+					uuidList.forEach(livingEntity.getCollidableExemptions()::remove));
+				break;
+		}
+	}
+
+}
```

Comment:
you should try to make it less repetitive

Reply:
> Out of curiosity how would you propose it?

so above the switch, you would do
```
 getExpr().stream(event).forEach(livingEntity -> {
        List<UUID> exemptions = livingEntity.getCollidableExemptions();
```

and then you could do `exemptions.addAll(uuidlist)` or `exemptions.clear` instead of redoing the stream everytime

my logic could also be completely wrong, but idk 🤷

Reply:
This is indeed applicable, which all left is the unexpected unforeseen bump ahead 😬